drivers/sensor/lsm6dsr/lsm6dsr.c

```diff
@@ -0,0 +1,803 @@
+	/* Convert to m/s^2 */
+	dval = (int64_t)raw_val * sensitivity * SENSOR_G_DOUBLE;
+	val->val1 = (int32_t)(dval / 1000000);
+	val->val2 = (int32_t)(dval % 1000000);
+
+}
+
+static inline int lsm6dsr_accel_get_channel(enum sensor_channel chan,
+					    struct sensor_value *val,
+					    struct lsm6dsr_data *data,
+					    uint32_t sensitivity)
+{
+	uint8_t i;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X: {
+		lsm6dsr_accel_convert(val, data->acc[0], sensitivity);
+	} break;
+	case SENSOR_CHAN_ACCEL_Y: {
+		lsm6dsr_accel_convert(val, data->acc[1], sensitivity);
+	} break;
+	case SENSOR_CHAN_ACCEL_Z: {
+		lsm6dsr_accel_convert(val, data->acc[2], sensitivity);
+	} break;
+	case SENSOR_CHAN_ACCEL_XYZ: {
+		for (i = 0; i < 3; ++i) {
+			lsm6dsr_accel_convert(val++, data->acc[i], sensitivity);
+		}
+	} break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int lsm6dsr_accel_channel_get(enum sensor_channel chan,
+				     struct sensor_value *val,
+				     struct lsm6dsr_data *data)
+{
+	return lsm6dsr_accel_get_channel(chan, val, data, data->acc_gain);
+}
+
+static inline void lsm6dsr_gyro_convert(struct sensor_value *val, int raw_val,
+					uint32_t sensitivity)
+{
+	int64_t dval;
+
+	/* Sensitivity is exposed in udps/LSB */
+	/* Convert to rad/s */
+	dval = (int64_t)raw_val * sensitivity * SENSOR_DEG2RAD_DOUBLE;
+	val->val1 = (int32_t)(dval / 1000000);
+	val->val2 = (int32_t)(dval % 1000000);
+}
+
+static inline int lsm6dsr_gyro_get_channel(enum sensor_channel chan,
+					   struct sensor_value *val,
+					   struct lsm6dsr_data *data,
+					   float sensitivity)
+{
+	uint8_t i;
+
+	switch (chan) {
+	case SENSOR_CHAN_GYRO_X: {
+		lsm6dsr_gyro_convert(val, data->gyro[0], sensitivity);
+	} break;
+	case SENSOR_CHAN_GYRO_Y: {
+		lsm6dsr_gyro_convert(val, data->gyro[1], sensitivity);
+	} break;
+	case SENSOR_CHAN_GYRO_Z: {
+		lsm6dsr_gyro_convert(val, data->gyro[2], sensitivity);
+	} break;
+	case SENSOR_CHAN_GYRO_XYZ: {
+		for (i = 0; i < 3; ++i) {
+			lsm6dsr_gyro_convert(val++, data->gyro[i], sensitivity);
+		}
+	} break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int lsm6dsr_gyro_channel_get(enum sensor_channel chan,
+				    struct sensor_value *val,
+				    struct lsm6dsr_data *data)
+{
+	return lsm6dsr_gyro_get_channel(chan, val, data, data->gyro_gain);
+}
+
+#if defined(CONFIG_LSM6DSR_ENABLE_TEMP)
+static void lsm6dsr_gyro_channel_get_temp(struct sensor_value *val,
+					  struct lsm6dsr_data *data)
+{
+	/* val = temp_sample / 256 + 25 */
+	val->val1 = data->temp_sample / 256 + 25;
+	val->val2 = (data->temp_sample % 256) * (1000000 / 256);
+}
+#endif /* defined(CONFIG_LSM6DSR_ENABLE_TEMP) */
+
+static int lsm6dsr_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct lsm6dsr_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ: {
```

Comment:
I think we prefer to not have `{ ... }` inside `case` statements when unnecessary.

There are no variables local to the majority of the `case` scopes I've seen here, so I would suggest removing the curly braces.